pkgs/apps/dirac/default.nix

```diff
@@ -0,0 +1,131 @@
+{ stdenv, lib, fetchurl, gfortran, cmake, makeWrapper
+, which, openssh, blas, lapack, mpi, boost, exatensor, python3
+} :
+
+assert
+  lib.asserts.assertMsg
+  (!blas.isILP64)
+  "A 32 bit integer implementation of BLAS is required.";
+
+stdenv.mkDerivation rec {
+  pname = "dirac";
+  version = "21.0";
+
+  nativeBuildInputs = [
+    which
+    gfortran
+    cmake
+    python3
+    makeWrapper
+    openssh
+  ];
+
+  # For now, DIRAC is built without the PCMSolver optional dependency.
+  buildInputs = [
+    blas
+    lapack
+    boost
+    exatensor
+  ];
+
+  propagatedBuildInputs = [
+    mpi
+  ];
+
+  src = fetchurl {
+    url = "https://zenodo.org/record/4836496/files/DIRAC-${version}-Source.tar.gz";
+    sha256 = "0vqmin24xqkcp3rvml0dimc4kjmb3m37c3l6bkjy4y6d3xyb41yg";
+  };
+
+  prePatch = ''
+    ls -lah
+  '';
+
+  patches = [
+    ./0001-exatensor-cmake.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace ./cmake/custom/exatensor.cmake \
+    --subst-var-by exatensor ${exatensor}
+
+    patchShebangs .
+  '';
+
+  FC = "mpif90";
+  CC = "mpicc";
+  CXX = "mpicxx";
```

Comment:
Does the project use C++?

Reply:
Yes, for some modules like in the `src/xcfun` package.

Reply:
XCFun is part of upstream nixpkgs. Can you use this one?

Reply:
@sheepforce The sources are shipped with Dirac (no git pull etc.). I don't think we need to replace them with upstream sources (that may just cause unnecessary trouble).

Reply:
I see. I thought those were just API wrappers for XCFun. :slightly_smiling_face: